configure-ironic.sh

```diff
@@ -45,3 +53,21 @@ EOF
 
 mkdir -p /shared/html
 mkdir -p /shared/ironic_prometheus_exporter
+
+if [ "$USE_HTTP_BASIC" = "true" ]; then
+
+crudini --merge /etc/ironic/ironic.conf <<EOF
+[DEFAULT]
+auth_strategy = http_basic
+http_basic_auth_user_file = $IRONIC_HTTP_BASIC_FILE
+
+[json_rpc]
+auth_strategy = http_basic
+http_basic_auth_user_file = $JSONRPC_HTTP_BASIC_FILE
+http_basic_username = $IRONIC_HTTP_BASIC_USERNAME
+http_basic_password = $IRONIC_HTTP_BASIC_PASSWORD
+EOF
+
+htpasswd -nbB $IRONIC_HTTP_BASIC_USERNAME $IRONIC_HTTP_BASIC_PASSWORD >> $IRONIC_HTTP_BASIC_FILE
+htpasswd -nbB $JSONRPC_HTTP_BASIC_USERNAME $JSONRPC_HTTP_BASIC_PASSWORD >> $JSONRPC_HTTP_BASIC_FILE
```

Comment:
If the basic auth files are in /shared and each container appends to it on startup won't we end up with accumulating duplicated entries over time (and possibly racy writes)?

Maybe it would be better to leave the default http_basic_auth_user_file as /etc/ironic/htpasswd and /etc/ironic/htpasswd-json-rpc. Then this startup script can overwrite (>) instead of append (>>) to these files.

Reply:
I will change for (>), based on the other comments I think we want /shared as the default so we won't provide an option to override...